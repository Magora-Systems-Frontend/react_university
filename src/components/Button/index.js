import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import './style.scss';

export default class Button extends Component {
  static propTypes = {
    children: PropTypes.any,
    onClick: PropTypes.func,
    type: PropTypes.string,
    modificationClass: PropTypes.string,
  };

  render() {
    const { children, onClick, colorStyle, cssModify, type, modificationClass } = this.props;
    const customClass = {
      bordered: 'btn_bordered',
      colored: 'btn_colored',
      transparent: 'btn_transparent',
      search: 'btn_search',
      mobileLogin: 'btn_mobile-login',
      category: 'btn_category-toggle',
    };
    return (
      <button
        className={cn('btn-action', customClass[colorStyle], modificationClass)}
        onClick={onClick ? onClick : () => {}}
        style={cssModify}
        type={type}>
        {children}
      </button>
    );
  }
}
