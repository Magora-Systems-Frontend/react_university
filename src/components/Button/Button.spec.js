import React from 'react';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Button from './index';

configure({ adapter: new Adapter() });

describe('button snapshot', () => {
  it('capturing snapshot of button', () => {
    const renderedValue = renderer.create(<Button type="button" />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('Shallow render Button component', () => {
  let wrapper;

  const objectProps = {
    children: 'string',
    onClick: (a) => a,
    type: 'button',
    colorStyle: 'colored',
    modificationClass: 'modClass',
  };

  beforeEach(() => {
    const { children, onClick, type, modificationClass, colorStyle } = objectProps;
    wrapper = shallow(
      <Button type={type} modificationClass={modificationClass} onClick={onClick} colorStyle={colorStyle}>
        {children}
      </Button>
    );
  });
  it('render component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('check children', () => {
    expect(wrapper.contains(objectProps.children)).toBe(true);
  });

  it('check type', () => {
    expect(wrapper.type(objectProps.type)).toBe(objectProps.type);
  });

  it('check modificationClass', () => {
    expect(wrapper.hasClass(objectProps.modificationClass)).toBe(true);
  });
  it('check colorStyle class', () => {
    expect(wrapper.hasClass(`btn_${objectProps.colorStyle}`)).toBe(true);
  });

  it('simulate click', () => {
    wrapper.simulate('click');
  });
});
