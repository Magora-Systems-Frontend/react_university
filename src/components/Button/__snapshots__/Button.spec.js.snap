// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snapshot capturing snapshot of button 1`] = `
<button
  className="btn-action"
  onClick={[Function]}
  type="button"
/>
`;
